vector-im/element-android
# Push notifications toggle: align implementation for current session (PSG-971)

## Type of change


## Content


Simplifying and fixing the logic for the push notifications toggle for the current session in "Settings -> Notifications".
The main update is to only rely on the related local preference. This preference is now updated whenever the pusher or account data related to the current device is edited as it may be remotely updated from another device (see `EnableNotificationsSettingUpdater`).
Unit tests have been added to cover all this logic.

## Motivation and context


Closes #7512 

## Screenshots / GIFs





## Tests



- Signin to server "synapse-prefill.lab.element.dev"
- Enable the new session manager labs setting
- Go to Settings -> Notifications
- Check the notifications are enabled for the current session
- Check notifications are well received on the device in this case
- Disable the notifications for the session
- Check notifications are no more received on the device
- Go to Settings -> Security & Privacy -> Show all sessions
- Go to the current session overview
- Check the notifications are disabled for the current session
- Enable notifications
- Check notifications are well received on the device in this case
- Go to Settings -> Notifications
- Check the notifications are enabled for the current session  

## Tested devices

- OS version(s): Android 11

## Checklist
